packages/filebrowser/src/listing.ts

```diff
@@ -2663,7 +2669,18 @@ export namespace DirListing {
       let modText = '';
       let modTitle = '';
       if (model.last_modified) {
-        modText = Time.formatHuman(new Date(model.last_modified));
+        // Render the date in one of multiple formats, depending on the container's size
+        const modifiedWidth = modified.getBoundingClientRect().width;
```

Comment:
Thank you for addressing my suggestion! This line has to be moved out of here. This is because right now the rendering sequence leads to layout trashing and opening a larger folder takes 3 seconds in rendering time alone:

![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/934f33fe-071c-4b61-8268-6cb6a2a66ac0)

Two ways to do that (possibly more):
- create a private `this._modifiedWidth` and record it in `onResize()`, or
- pass it down as an argument to `updateItemNode()`

Reply:
Recent changes only rerender the modified date elements, and no other elements of the file listings, when a resize event causes the date style to change.